hws/mlx_hw_steering_parser.py

```diff
@@ -244,30 +248,33 @@ def parse_args():
 
     _config_args["extra_hw_res_arg"] = False
     _config_args["extra_hw_res_pat"] = False
+    _config_args["extra_hw_res_counter"] = False
 
     if (args.dump_hw_resources):
         _config_args["dump_hw_resources"] = True
         if (args.device == ""):
             _config_args["device"] = None
         else:
             _config_args["device"] = args.device
-
-        for hw_res in args.extra_hw_res.split(","):
-            if hw_res == "all":
-                _config_args["extra_hw_res_pat"] = True
-                _config_args["extra_hw_res_arg"] = True
-                break
-            elif hw_res == "pat":
-                _config_args["extra_hw_res_pat"] = True
-            elif hw_res == "arg":
-                _config_args["extra_hw_res_arg"] = True
-
-        if _config_args.get("extra_hw_res_arg") and not(_config_args.get("extra_hw_res_pat")):
-            _config_args["extra_hw_res_arg"] = False
-
     else:
         _config_args["dump_hw_resources"] = False
 
+    for hw_res in args.extra_hw_res.split(","):
+        if hw_res == "all":
+            _config_args["extra_hw_res_pat"] = True
+            _config_args["extra_hw_res_arg"] = True
+            _config_args["extra_hw_res_counter"] = True
+            break
+        elif hw_res == "pat":
+            _config_args["extra_hw_res_pat"] = True
+        elif hw_res == "arg":
+            _config_args["extra_hw_res_arg"] = True
+        elif hw_res == "counter":
+            _config_args["extra_hw_res_counter"] = True
+
+    if _config_args.get("extra_hw_res_arg") == True and not(_config_args.get("extra_hw_res_pat")):
+        _config_args["extra_hw_res_arg"] = False
```

Comment:
This logic was present in the script before this patch, so it's not something new, but still, question: so you check if the user asked for args w/o asking for patterns, and then you just silently ignoring the args part?

Reply:
yes
will provide different patch to output message